src/presentation/controllers/device-controller.ts

```diff
@@ -0,0 +1,86 @@
+import {BaseController, GetRequest, PostRequest, RestController} from './base-controller'
+import {Request, Response} from 'express'
+import {DeviceService} from '../../business-logic/services/interfaces/device-service'
+import {HttpErrorHandler} from '../interfaces/http-error-handler'
+import {Exception} from '../../model/exceptions/exception'
+import {HttpResponseFormatter} from '../interfaces/http-response-formatter'
+import {Device, INewsDevice, TelemetricsDevice} from '../../model/entities/device'
+import {DeviceType} from '../../model/enums/device-type'
+import {DeviceDtoInterface, INewsDeviceDto, TelemetricsDeviceDto} from '../dtos/device-dto'
+import {HttpStatusCode} from '../http-status-code'
+
+
+@RestController('/devices')
+export class DeviceController extends BaseController {
+
+  constructor(
+    private readonly deviceService: DeviceService,
+    private readonly httpErrorHandler: HttpErrorHandler,
+    private readonly httpResponseFormatter: HttpResponseFormatter,
+  ) {
+    super()
+  }
+
+  @GetRequest()
+  public async getAllDevices(_request: Request, response: Response): Promise<void> {
+    try {
+      const devices: Device[] = await this.deviceService.getDevices()
+      response.send(this.httpResponseFormatter.formatSuccessResponse(
+        devices.map(device => this.toDeviceDto(device))
+      ))
+    } catch (error) {
+      this.httpErrorHandler.handleError(response, error as Exception)
+    }
+  }
+
+  @GetRequest('/:deviceId')
+  public async getDevice(request: Request, response: Response): Promise<void> {
+    try {
+      const {deviceId} = request.params
```

Comment:
If i have to follow the conventions that the guys use in other places, then I have to suggest this 😄  :

```suggestion
      const deviceId: string = request.params.deviceId
```